src/compiler/parser.ts

```diff
@@ -5139,8 +5139,26 @@ namespace Parser {
     }
 
     function nextTokenIsIdentifierOnSameLine() {
+        const isTypeKeyword = token() === SyntaxKind.TypeKeyword;
         nextToken();
-        return !scanner.hasPrecedingLineBreak() && isIdentifier();
+
+        if (scanner.hasPrecedingLineBreak()) {
+            return false;
+        }
+
+        if (!isTypeKeyword) {
+            return isIdentifier();
+        }
+
+        switch (token()) {
+            case SyntaxKind.AsKeyword:
+            case SyntaxKind.SatisfiesKeyword: {
+                return lookAhead(() => nextToken() === SyntaxKind.EqualsToken);

```

Comment:
~~The name of a type alias isn't always immediately followed by an `=`, so I don't think this specific look-ahead is enough.~~

See the comment I'm leaving in my review.

```ts
type satisfies<T> = T;
```